src/platform/silabs/wifi/SiWx/WifiInterfaceImpl.cpp

```diff
@@ -939,10 +876,63 @@ CHIP_ERROR WifiInterfaceImpl::StartNetworkScan(chip::ByteSpan ssid, ::ScanCallba
     }
     wfx_rsi.scan_cb = callback;
 
-    // TODO: We should be calling the start function directly instead of doing it asynchronously
-    WifiPlatformEvent event = WifiPlatformEvent::kScan;
-    PostWifiPlatformEvent(event);
+    sl_status_t status = SL_STATUS_OK;
 
+    sl_wifi_scan_configuration_t wifi_scan_configuration = default_wifi_scan_configuration;
+    if (wfx_rsi.dev_state.Has(WifiInterface::WifiState::kStationConnected))
+    {
+        /* Terminate with end of scan which is no ap sent back */
+        wifi_scan_configuration.type                   = SL_WIFI_SCAN_TYPE_ADV_SCAN;
+        wifi_scan_configuration.periodic_scan_interval = kAdvScanPeriodicity;
+    }
+
+    sl_wifi_advanced_scan_configuration_t advanced_scan_configuration = {
+        .trigger_level        = kAdvScanThreshold,
+        .trigger_level_change = kAdvRssiToleranceThreshold,
+        .active_channel_time  = kAdvActiveScanDuration,
+        .passive_channel_time = kAdvPassiveScanDuration,
+        .enable_instant_scan  = kAdvEnableInstantbgScan,
+        .enable_multi_probe   = kAdvMultiProbe,
+    };
+
+    status = sl_wifi_set_advanced_scan_configuration(&advanced_scan_configuration);
+
+    // TODO: Seems like ChipDie should be called here, the device should be initialized here
+    VerifyOrReturnError(
+        status == SL_STATUS_OK, CHIP_ERROR_INTERNAL,
+        ChipLogError(DeviceLayer, "sl_wifi_set_advanced_scan_configuration failed: 0x%lx", static_cast<uint32_t>(status)));
+
+    sl_wifi_set_scan_callback(BackgroundScanCallback, nullptr);
+    wfx_rsi.dev_state.Set(WifiInterface::WifiState::kScanStarted);
+
+    // If an ssid was not provided, we need to call the scan API with nullptr to scan all Wi-Fi networks
+    sl_wifi_ssid_t _ssid     = { 0 };
+    sl_wifi_ssid_t * ssidPtr = nullptr;
```

Comment:
This whole logic here seems fishy 🐟  why do we need to go throught all of the steps above ? Does the `sl_wifi_start_scan` actually need a string to an SSID or not ? because we seem to call it even if ssidPtr is null...